bob/src/cluster/quorum.rs

```diff
@@ -32,33 +32,34 @@ impl Quorum {
 
     async fn put_at_least(&self, key: BobKey, data: &BobData) -> Result<(), Error> {
         let mut local_put_ok = 0_usize;
-        let at_least = self.quorum;
+        let mut at_least = self.quorum;
         let mut failed_nodes = Vec::new();
-        let (vdisk_id, disk_path) = self.mapper.get_operation(key);
-        let (tasks, errors) =
-            if let Some(disk_path) = disk_path {
-                debug!("PUT[{}] ~~~PUT TO {} REMOTE NODES AND LOCAL NODE~~~", key, at_least - 1);
-                let (remote_put, local_put) = tokio::join!(
-                    self.put_remote_nodes(key, data, at_least - 1),
-                    put_local_node(&self.backend, key, data, vdisk_id, disk_path),
-                );
-                let (mut remote_tasks, mut errors) = remote_put;
-                if let Err(e) = local_put {
-                    error!("{}", e);
-                    failed_nodes.push(self.mapper.local_node_name().clone());
-                    debug!("PUT[{}] local failed, put another remote", key);
-                    errors.extend(finish_at_least_handles(&mut remote_tasks, 1).await.into_iter());
-                } else {
-                    local_put_ok += 1;
-                    debug!("PUT[{}] local node put successful", key);
+        let (vdisk_id, disk_paths) = self.mapper.get_operation(key);
+        let affected_replicas_by_node = self.mapper.get_replicas_count_by_node(key);
+        let (tasks, errors) = 
+            if let Some(paths) = disk_paths {
+                let paths_len = paths.len();
+                debug!("PUT[{}] ~~~PUT TO {} REMOTE NODES AND LOCAL NODE~~~", key, at_least - paths_len);
+                let ((tasks, errors), local_puts) = tokio::join!(
+                    self.put_remote_nodes(key, data, at_least, &affected_replicas_by_node),
+                    put_local_node_all(&self.backend, key, data, vdisk_id, paths));
+                if local_puts != paths_len {
+                    failed_nodes.push(self.mapper.local_node_name().to_owned());
                 }
-                (remote_tasks, errors)
+                local_put_ok += local_puts;
+                at_least -= local_puts;
```

Comment:
Don't see a reason to update `at_least`, because later it only used once id `debug` log, where the same value can be calculated on the fly (`at_least - local_puts`)

Reply:
We don't have `local_puts` later, it's inside if scope, I think it's better to leave modification here

Reply:
But you have `local_put_ok`, that equals to `local_puts`